Mathlib/GroupTheory/GroupAction/Basic.lean

```diff
@@ -533,6 +533,102 @@ theorem orbitRel.Quotient.orbit_eq_orbit_out (x : orbitRel.Quotient G α)
+  orbitRel.quotient_eq_of_quotient_subgroup_eq h
+
+@[to_additive]
+nonrec lemma orbitRel.Quotient.orbit_nonempty (x : orbitRel.Quotient G α) :
+    Set.Nonempty x.orbit := by
+  rw [orbitRel.Quotient.orbit_eq_orbit_out x Quotient.out_eq']
+  exact orbit_nonempty _
+
+@[to_additive]
+nonrec lemma orbitRel.Quotient.mapsTo_smul_orbit (g : G) (x : orbitRel.Quotient G α) :
+    Set.MapsTo (g • ·) x.orbit x.orbit := by
+  rw [orbitRel.Quotient.orbit_eq_orbit_out x Quotient.out_eq']
+  exact mapsTo_smul_orbit g x.out'
+
+@[to_additive]
+instance (x : orbitRel.Quotient G α) : MulAction G x.orbit where
+  smul g := (orbitRel.Quotient.mapsTo_smul_orbit g x).restrict _ _ _
+  one_smul a := Subtype.ext (one_smul G (a : α))
+  mul_smul g g' a' := Subtype.ext (mul_smul g g' (a' : α))
+
+@[to_additive (attr := simp)]
+lemma orbitRel.Quotient.orbit.coe_smul {g : G} {x : orbitRel.Quotient G α} {a : x.orbit} :
+    ↑(g • a) = g • (a : α) :=
+  rfl
+
+@[to_additive]
+instance (x : orbitRel.Quotient G α) : IsPretransitive G x.orbit where
+  exists_smul_eq := by
+    induction x using Quotient.inductionOn'
+    rintro ⟨y, yh⟩ ⟨z, zh⟩
+    rw [orbitRel.Quotient.mem_orbit, Quotient.eq''] at yh zh
+    rcases yh with ⟨g, rfl⟩
+    rcases zh with ⟨h, rfl⟩
+    refine ⟨h * g⁻¹, ?_⟩
+    ext
+    simp [mul_smul]
+
+@[to_additive]
+lemma orbitRel.Quotient.mem_subgroup_orbit_iff {H : Subgroup G} {x : orbitRel.Quotient G α}
+    {a b : x.orbit} : a ∈ MulAction.orbit H b ↔ (a : α) ∈ MulAction.orbit H (b : α) := by
+  refine ⟨fun h ↦ ?_, fun h ↦ ?_⟩
+  · rcases h with ⟨g, rfl⟩
+    simp_rw [Submonoid.smul_def, Subgroup.coe_toSubmonoid, orbit.coe_smul, ← Submonoid.smul_def]
+    exact MulAction.mem_orbit _ g
+  · rcases h with ⟨g, h⟩
+    simp_rw [Submonoid.smul_def, Subgroup.coe_toSubmonoid, ← orbit.coe_smul,
+             ← Submonoid.smul_def, ← Subtype.ext_iff] at h
+    subst h
+    exact MulAction.mem_orbit _ g
+
+@[to_additive]
+lemma orbitRel.Quotient.subgroup_quotient_eq_iff {H : Subgroup G} {x : orbitRel.Quotient G α}
+    {a b : x.orbit} : (⟦a⟧ : orbitRel.Quotient H x.orbit) = ⟦b⟧ ↔
+      (⟦↑a⟧ : orbitRel.Quotient H α) = ⟦↑b⟧ := by
+  simp_rw [← @Quotient.mk''_eq_mk, Quotient.eq'']
+  exact orbitRel.Quotient.mem_subgroup_orbit_iff
+
+@[to_additive]
+lemma orbitRel.Quotient.mem_subgroup_orbit_iff' {H : Subgroup G} {x : orbitRel.Quotient G α}
+    {a b : x.orbit} {c : α} (h : (⟦a⟧ : orbitRel.Quotient H x.orbit) = ⟦b⟧) :
+    (a : α) ∈ MulAction.orbit H c ↔ (b : α) ∈ MulAction.orbit H c := by
+  simp_rw [mem_orbit_symm (a₂ := c)]
+  convert Iff.rfl using 2
+  rw [orbit_eq_iff]
+  suffices hb : ↑b ∈ orbitRel.Quotient.orbit (⟦a⟧ : orbitRel.Quotient H x.orbit) by
+    rw [orbitRel.Quotient.orbit_eq_orbit_out (⟦a⟧ : orbitRel.Quotient H x.orbit) Quotient.out_eq']
+       at hb
+    rw [← orbitRel.Quotient.mem_subgroup_orbit_iff]
+    convert hb using 1
+    rw [orbit_eq_iff]
+    change @Setoid.r _ (orbitRel H _) _ _
```

Comment:
I've added `orbitRel_r_apply`.

In general I may not know the best way to make use of the API for quotients; I seem to find a lot that I have things in one form (`Setoid.r`, `Setoid.rel`, something defeq to those for the particular setoid, `≈`, ..., or `mk''` versus `mk` for quotients), and need to keep changing to a different form because the particular lemma I want to use next is only available for some of the forms, and also often find that key information is implicit and not shown in the infoview (making it hard to follow the infoview state, which e.g. just says `Setoid.r` with no information about which setoid is involved, as well as often requiring use of `@`), especially when, as here, dealing with multiple quotients for the same type (orbits for a group and a subgroup thereof, in this case).